llvm/lib/DebugInfo/GSYM/DwarfTransformer.cpp

```diff
@@ -596,29 +612,26 @@ Error DwarfTransformer::convert(uint32_t NumThreads, raw_ostream *OS) {
       DWARFDie Die = getDie(*CU);
       if (Die) {
         CUInfo CUI(DICtx, dyn_cast<DWARFCompileUnit>(CU.get()));
-        pool.async([this, CUI, &LogMutex, OS, Die]() mutable {
-          std::string ThreadLogStorage;
-          raw_string_ostream ThreadOS(ThreadLogStorage);
-          handleDie(OS ? &ThreadOS: nullptr, CUI, Die);
-          ThreadOS.flush();
-          if (OS && !ThreadLogStorage.empty()) {
-            // Print ThreadLogStorage lines into an actual stream under a lock
-            std::lock_guard<std::mutex> guard(LogMutex);
-            *OS << ThreadLogStorage;
-          }
+        pool.async([this, CUI, &LogMutex, Out, Die]() mutable {
+          StringAggregator ThreadOut(Out.IsShowingDetail());
+          handleDie(ThreadOut, CUI, Die);
+          // Print ThreadLogStorage lines into an actual stream under a lock
+          std::lock_guard<std::mutex> guard(LogMutex);
+          Out.Merge(ThreadOut);
         });
       }
     }
     pool.wait();
   }
   size_t FunctionsAddedCount = Gsym.getNumFunctionInfos() - NumBefore;
-  if (OS)
-    *OS << "Loaded " << FunctionsAddedCount << " functions from DWARF.\n";
+  if (Out.IsShowingDetail())
```

Comment:
```
if (Out.GetOS())
```
or we can rely on the fact that the `Out` object will not foward the info if there is no stream by just writing:
```
Out << "Loaded " << FunctionsAddedCount << " functions from DWARF.\n";
```
(no need to check if there is a stream because `template <typename T> OutputAggregator &operator<<(T &&value)` checks the stream and we aren't putting anything too expensive into the stream

Reply:
👍